### PR TITLE
Add basic KATCPSensensor.wait()

### PR DESCRIPTION
**Pull request by brickZA**
Also implement KATCPResource.wait() in usingKATCPSensor.wait()

### CODE REVIEW

File: katcp/resource.py [hunk 1]
```diff
@@ -690,6 +669,72 @@ def get_status(self):
...
+                    ioloop.add_callback(f.set_result, True)
+            except Exception:
+                f.set_exc_info(sys.exc_info())
+                self.unregister_listener(handle_update)
+
+        self.register_listener(handle_update)
+        ioloop.add_callback(handle_update, self, self._reading)
+
+        if timeout:
+            to = ioloop.time()+timeout
```
Comment (theunsa) [inline katcp/resource.py]: pep8 ... spaces around '+'

File: katcp/resource.py [hunk 2]
```diff
@@ -690,6 +669,72 @@ def get_status(self):
...
+                status_matched = reading.status == status or status is None
+                if val_matched and status_matched:
+                    self.unregister_listener(handle_update)
+                    ioloop.add_callback(f.set_result, True)
+            except Exception:
+                f.set_exc_info(sys.exc_info())
+                self.unregister_listener(handle_update)
+
+        self.register_listener(handle_update)
+        ioloop.add_callback(handle_update, self, self._reading)
```
Comment (theunsa) [inline katcp/resource.py]: Question? The ioloop.add_callback line will result in 'handle_update' being called on the next ioloop iteration ... so it is not to say that the sensor had an updated value yet. Where is the actual delay to allow the conditions to be met within 'timeout' time?

Comment (theunsa) [inline katcp/resource.py]: Question2? What happens if the sensor gets an updated value before the next ioloop iteration gets to call 'handle_update'?

File: katcp/resource.py (hunk 1 above)
Comment (brickZA) [inline katcp/resource.py]: Done

**Pull request merged**